client/customer-effort-score-tracks/customer-effort-score-tracks-container.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import { compose } from '@wordpress/compose';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import CustomerEffortScoreTracks from './customer-effort-score-tracks';
+
+/**
+ * Maps the queue of CES tracks surveys to CustomerEffortScoreTracks
+ * components. Note that generally there will only be a single survey per page
+ * however this is designed to be flexible if multiple surveys per page are
+ * added in the future.
+ *
+ * @param {Object}   props                      Component props.
+ * @param {Array}    props.queue                The queue of surveys.
+ * @param {boolean}  props.resolving            Whether the queue is resolving.
+ * @param {Function} props.clearQueueFromOption Sets up clearing of the queue on the next page load.
+ */
+function CustomerEffortScoreTracksContainer( {
+	queue,
+	resolving,
+	clearQueueFromOption,
+} ) {
+	if ( resolving ) {
+		return null;
+	}
+
+	if ( queue.length ) {
+		clearQueueFromOption();
+	}
+
+	return (
+		<>
+			{ queue.map( ( item, index ) => (
+				<CustomerEffortScoreTracks
+					key={ index }
+					initiallyVisible={ true }
+					trackName={ item.track_name }
+					label={ item.label }
+					trackProps={ item.props || {} }
+				/>
+			) ) }
+		</>
+	);
+}
+
+CustomerEffortScoreTracksContainer.propTypes = {
+	/**
+	 * The queue of CES tracks surveys to display.
+	 */
+	queue: PropTypes.arrayOf( PropTypes.object ),
+	/**
+	 * If the queue option is being resolved.
+	 */
+	resolving: PropTypes.bool,
+	/**
+	 * Set up clearing the queue on the next page load.
+	 */
+	clearQueueFromOption: PropTypes.func,
```

Comment:
The `FromOption` portion of this function name spells out some specifics of the implementation. I don't think it lends any clarity to the purpose of the method and could be potentially confusing should that implementation end up changing later on.

Reply:
sounds good, I wasn't entirely happy with this naming

Reply:
fixed in 3db8d9a